src/main/java/pages/cartPage/shoppingCartTable/ShoppingCartTableRow.java

```diff
@@ -0,0 +1,199 @@
+    }
+
+    public BigDecimal getUnitPrice(String productName) {
+        int rowNumber = getRowNumber(productName);
+        return getUnitPrice(rowNumber);
+    }
+
+    public BigDecimal getUnitPrice(ProductModel product) {
+        int rowNumber = getRowNumber(product.getName());
+        return getUnitPrice(rowNumber);
+    }
+
+    public BigDecimal getTotalPrice(int rowNumber) {
+        var row = getRowContent(rowNumber).get(ShoppingCartHeader.TOTAL.columnIndex);
+        return new BigDecimal(row.substring(1));
+    }
+
+    public BigDecimal getTotalPrice(String productName) {
+        int rowNumber = getRowNumber(productName);
+        return getTotalPrice(rowNumber);
+    }
+
+    public BigDecimal getTotalPrice(ProductModel product) {
+        int rowNumber = getRowNumber(product.getName());
+        return getTotalPrice(rowNumber);
+    }
+
+    public void setQuantity(int rowNumber, int quantity) {
+        var row = getRow(rowNumber);
+        var input = row.findElements(By.tagName("td"))
+                .get(ShoppingCartHeader.QUANTITY.columnIndex)
+                .findElement(quantityItemInput);
+        input.clear();
+        input.sendKeys(String.valueOf(quantity));
+    }
+
+    public void setQuantity(String productName, int quantity) {
+        int rowNumber = getRowNumber(productName);
+        setQuantity(rowNumber, quantity);
+    }
+
+    public void setQuantity(ProductModel product, int quantity) {
+        int rowNumber = getRowNumber(product.getName());
+        setQuantity(rowNumber, quantity);
+    }
+
+    public int getQuantity(int rowNumber) {
+        return Integer.parseInt(getQuantityInputValue(getRow(rowNumber)));
+    }
+
+    public int getQuantity(String productName) {
+        int rowNumber = getRowNumber(productName);
+        return getQuantity(rowNumber);
+    }
+
+    public int getQuantity(ProductModel product) {
+        int rowNumber = getRowNumber(product.getName());
+        return getQuantity(rowNumber);
+    }
+
+    private String getQuantityInputValue(WebElement cell) {
+        return cell.findElements(quantityItemInput).getFirst().getAttribute("value");
+    }
+
+    private String getImageItemSrc(WebElement cell) {
+        return cell.findElements(By.xpath(".//img")).getFirst().getAttribute("src");
+    }
+
+    private String getDeleteButtonHref(WebElement cell) {
+        return cell.findElement(By.xpath("./a")).getAttribute("href");
+    }
+
+    public void deleteProduct(int rowNumber) {
+        WebElement row = getRow(rowNumber);
+        row.findElement(deleteItemButton).click();
+    }
+
+    public void deleteProduct(String productName) {
+        int rowNumber = getRowNumber(productName);
+        deleteProduct(rowNumber);
+    }
+
+    public void deleteProduct(ProductModel product) {
+        int rowNumber = getRowNumber(product.getName());
+        deleteProduct(rowNumber);
+    }
+
+    private List<WebElement> getTableRows() {
```

Comment:
I think we can also remove that one :)

Reply:
Thanks to the changes between ```ShoppingCartTable``` and ```ShoppingCartTableRow``` I can delete unnecessary methods, thanks!